services/src/main/java/org/fao/geonet/api/selections/FavouriteMetadataListApi.java

```diff
@@ -0,0 +1,682 @@
+    @RequestMapping(
+        method = RequestMethod.GET,
+        produces = {
+            MediaType.APPLICATION_JSON_VALUE
+        })
+    public
+    @ResponseBody
+    @ResponseStatus(HttpStatus.OK)
+    List<FavouriteMetadataListVM> getFavouritesLists(
+        @Parameter(hidden = true)
+        HttpSession httpSession,
+
+        @Parameter(hidden = true)
+            HttpServletRequest  httpServletRequest
+    )
+        throws Exception {
+        String sessionId = getSessionId(httpServletRequest);
+        User user =  getUser(httpSession);
+        boolean isAdmin = isAdmin(httpSession);
+        return favouriteMetadataListRepository.findPublic(user,sessionId)
+            .stream()
+            .map(x->new FavouriteMetadataListVM(x
+                ,permittedWrite(x,user,sessionId,isAdmin),
+                ownedByMe(x,user,sessionId)))
+            .collect(Collectors.toList());
+    }
+
+    /**
+     *   Get the `SESSION_COOKIE_NAME` cookie value from the request.
+     *   If there isn't one, then returns null.
+     */
+    String getSessionId(HttpServletRequest request) {
+        Cookie[] cookies = request.getCookies();
+        if (cookies ==null) {
+            return null;
+        }
+        Optional<Cookie> sessionCookie = Arrays.stream(cookies)
+            .filter(x->x.getName().equals(SESSION_COOKIE_NAME))
+            .findFirst();
+
+        return sessionCookie.isPresent() ?  sessionCookie.get().getValue() : null;
+    }
+
+    /**
+     *   adds a new SESSION_COOKIE_NAME cookie (set-cookie) to the response.
+     *   Value will be a random UUID.
+     *
+     * @return the SESSION_COOKIE_NAME value (uuid)
+     */
+    String setSessionId(HttpServletResponse response) {
+        return setSessionId(response, UUID.randomUUID().toString());
+    }
+
+    /**
+     *   adds a new SESSION_COOKIE_NAME cookie (set-cookie) to the response.
+     *
+     * @return the SESSION_COOKIE_NAME value (uuid)
+     */
+    String setSessionId(HttpServletResponse response, String value) {
+        response.addCookie(new Cookie(SESSION_COOKIE_NAME,value));
+        return value;
+    }
+
+    /**
+     *  Get a FavouriteMetadataListVM by id.
+     */
+    @io.swagger.v3.oas.annotations.Operation(summary = "Get a specific (by DB id) of FavouriteList list")
+    @RequestMapping(
+        method = RequestMethod.GET,
+        value = "/{favouriteListIdentifier}",
+        produces = {
+            MediaType.APPLICATION_JSON_VALUE
+        })
+    public
+    @ResponseBody
+    @ResponseStatus(HttpStatus.OK)
+    FavouriteMetadataListVM getFavouritesList(
+        @Parameter(description = "FavouritesList DB id (int)",required = true)
+        @PathVariable
+        Integer favouriteListIdentifier,
+        @Parameter(hidden = true)
+        HttpSession httpSession,
+        @Parameter(hidden = true)
+        HttpServletRequest  httpServletRequest
+    )
```

Comment:
the `sessionId` can be retrieved using the `@CookieValue` annotation:
```suggestion
    public
    @ResponseBody
    @ResponseStatus(HttpStatus.OK)
    FavouriteMetadataListVM getFavouritesList(
        @CookieValue(SESSION_COOKIE_NAME) sessionId,
        @Parameter(description = "FavouritesList DB id (int)",required = true)
        @PathVariable
        Integer favouriteListIdentifier,
        @Parameter(hidden = true)
        HttpSession httpSession,
        @Parameter(hidden = true)
        HttpServletRequest  httpServletRequest
    )
```